### ISSUE TITLE
Rename JsonTracer as ChromeTracer

### ISSUE THREAD

**Issue by dayo09** [Samsung/ONE-vscode#1183]
(just a suggestion, not about this PR)

It seems the name 'JsonTracer' is a bit confusing because we have another 'JsonTracer' module : mondrianViewer. Mondrian files are another type of our 'json' file which 'traces' mondrian (its file extension names are 'tracealloc.json').

Therefore, I suggest changing the module name as 'ChromeTracer'. We actually uses the 'chrome tracer' open source solution here, so the naming sounds more direct.

_Originally posted by @dayo09 in https://github.com/Samsung/ONE-vscode/pull/1172#discussion_r938619285_

**Comment by llFreetimell** [comment]
I agree with the background of this suggestion and also agree with the confusion :)

Only one thing I worry about is... "Can we really use the name of `Chrome` or `ChromeTracer` for our tool?" in legally.
Since `Chrome` and `ChromeTracer` is a kind of trademark of Google's, we may invade their right of the naming :(

So how about naming with another idea?
`OneTracer`, `CycleViewer`, `TimelineViewer`, etc.... nothing good idea comes now T.T

**Comment by dayo09** [comment]
@llFreetimell 

> We actually uses the 'chrome tracer' open source solution here, so the naming sounds more direct.

Well how do you think about the line I wrote above..?
We actually vulcanized their open source into our project, so I assumed that altering the name invades their efforts.

**Comment by llFreetimell** [comment]
> We actually vulcanized their open source into our project, so I assumed that altering the name invades their efforts.

If we embedded all or some part of `ChromeTracer`, I agree with you :)

But as `JsonTracer` is sort of copycat, which only referenced overall result of Chrome Tracer, I don't think this is sort of `ChromeTracer`. (Actually, `JsonTracer` did not copy source codes in `ChromeTracer`)

In addition, if we name it as `ChromeTracer`, people may misunderstand that we fully supports `ChromeTracer` features in ONE-vscode.
Or our own features which are not in `ChromeTracer` may be implemented in `JsonTracer` in the future.

But I totally agree that we need another name for `Jsontracer`...!

**Comment by dayo09** [comment]
@llFreetimell I thought the codes are from the catapult-project's tracing module..? https://github.com/catapult-project/catapult/tree/444aba89e1c30edf348c611a9df79e2376178ba8/tracing

**Comment by dayo09** [comment]
But we should keep in mind that our trace generation command is `--save-chrome-trace`.

**Comment by dayo09** [comment]
### Let's list up all the candidates then.

- Json Tracer
- Json Trace Viewer
- Trace Viewer
  
- Chrome Trace Viewer
 
- Latency Trace Viewer
- Performance Trace Viewer
- Performance Viewer


IMO, 
- "Json Tracer, Json Trace Viewer, Trace Viewer" sounds confusing along with mondrian, which uses the extension *.tracealloc.json.
- " Chrome Trace Viewer" sounds fine but with some objection about using 'chrome' naming.
- "Latency Trace Viewer", "Performance Trace Viewer", "Performance Viewer" may be another alternatives.

@ejjeong Could you give your opinion, too? It will be very helpful :-D
FYI, JsonTracer is our visualization module of chrome trace files. Find [this gif](https://github.com/Samsung/ONE-vscode/pull/1190#issuecomment-1207697398) to get the basic idea :-D.

**Comment by ejjeong** [comment]
> @ejjeong Could you give your opinion, too? It will be very helpful :-D

Ah, sorry I missed this comment 😢
I turned off the notifications from this repo (just because it is very active 😄).

I don't have any strong preference among the current candidates,
especially because I'm not aware of which "open source project" JsonTracer is based on.

Anyway, I'd like to add another candidate: "trace event viewer"
The backend profiler generates the trace in [Trace Event Format](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview).

**Comment by ejjeong** [comment]
BTW, just out of curiosity, was there any reason that `JsonTracer` did not use the catapult [`Trace-Viewer`](https://github.com/catapult-project/catapult/blob/2417ba3d0d766848fcd0bc65d6cf2819f68c16e5/tracing/README.md)?
It even provides [trace2html](https://github.com/catapult-project/catapult/blob/2417ba3d0d766848fcd0bc65d6cf2819f68c16e5/tracing/bin/trace2html), whose output should be easy to be shown in the vscode.

Ah, maybe is it because it's hard to run python code in vscode extentions?

**Comment by dayo09** [comment]
> Anyway, I'd like to add another candidate: "trace event viewer"

Thanks for giving your opinion! :-D

**Comment by llFreetimell** [comment]
Sorry for missing this issue... :(

I am confused now because I regared [Trace-Viewer](https://github.com/catapult-project/catapult/blob/2417ba3d0d766848fcd0bc65d6cf2819f68c16e5/tracing/README.md) as Chrome Tracer....
Maybe there are something I missed... T.T
So I want to withdraw my suggestions and objections 😅 

> was there any reason that JsonTracer did not use the catapult [Trace-Viewer](https://github.com/catapult-project/catapult/blob/2417ba3d0d766848fcd0bc65d6cf2819f68c16e5/tracing/README.md)?

JsonTracer referenced [Trace-Viewer](https://github.com/catapult-project/catapult/blob/2417ba3d0d766848fcd0bc65d6cf2819f68c16e5/tracing/README.md), which I called as Chrome Tracer... T.T
https://github.com/Samsung/ONE-vscode/blob/727a52a776478fe373996a64587bb50a89999523/media/Jsontracer/bar.js#L45-L46
